packages/spor-icon-elm/generate.js

```diff
@@ -24,52 +33,208 @@ async function generateSvgs() {
                     };
                 });
 
             const svgs = await Promise.all(svgPromises);
-            const svgNames = svgs.map((svg) => svg.name);
-            const svgImpls = svgs.map((svg) => svg.impl);
+            return svgs
+        })
+    await Promise.all(svgs).then(svgs => {
+        const gatheredSvgs = svgs.flat()
+        const outlines = gatheredSvgs.filter(svg => svg.type === "outline")
+        const fills = gatheredSvgs.filter(svg => svg.type === "fill")
+        const maxSizeFills = Object.values(groupBy(fills, "name")).map(group => maxBy(group, "sizeAsInt")) 
+        const maxSizeOutlines = Object.values(groupBy(outlines, "name")).map(group => maxBy(group, "sizeAsInt"))
+        
+        // Autogenerated elm svgs not to be exposed to end-user. To be appended to bottom of elm-filme and used in toHtml.
+        const svgImpls = [...maxSizeFills, ...maxSizeOutlines].map((svg) => svg.impl)
+
+        // "Allowed" icon sizes based on existing file names.
+        const uniqueIconSizes = Array.from(new Set([...gatheredSvgs.map(svg => svg.size)]))
+
+        const uniqueNames = Object.keys(groupBy(gatheredSvgs, "name")).map(camelCase)
+
+        const moduleHeader = generateModuleHeader("Icons", uniqueNames.sort(), uniqueIconSizes);
+        const src = [moduleHeader].concat(svgImpls)
+            .join('\n\n\n{-|-}\n');
+
+        const targetFile = path.join(targetFolder, `Icons.elm`);
+
+        fs.writeFile(targetFile, src);
+    })
+    
+}
 
-            const moduleHeader = generateModuleHeader(module.name, svgNames);
-            const src = [moduleHeader].concat(svgImpls)
-                .join('\n\n\n{-|-}\n');
+// Most file names are structured as "name-[fill|outline]-[size]x[size].svg"
+// If we don't have either fill or outline, we just assign one of them as "placeholder"-type.
+// Using the icon with any of the two types will still return the same icon.
+function parseFileName(fileName) {
+    const [iconName, fillType, size] = fileName.split("-")
+    const sizeAsNumber = size ? size.split("x")[0] : "24"
+    const fillTypeWithDefault = fillType === "outline" || fillType === "fill" ? fillType : "outline"
+    const sizeWithDefault = isNaN(sizeAsNumber) ? "24" : sizeAsNumber
+    return {
+        iconName,
+        fillType: fillTypeWithDefault,
+        size: sizeWithDefault
+    }
+}
 
-            return {
-                ...module,
-                src: src
-            };
-        }).forEach(async (modulePromise) => {
-            const module = await modulePromise;
+function injectUnderscoreToImpl(implString, name) {
+    return implString.replaceAll(name, name + "_")
+}
+
+// Default fill color is set on parent with Elm, to allow differentiation on backgrounds, etc.
+function removeDefaultFillColor(svgString) {
+    return svgString.replaceAll('fill="#2B2B2C"', "")
+}
+
+function removeHardcodedSize(svgString) {
+    return svgString.replace(/width="\d{2}"/,"").replace(/height="\d{2}"/, "")
```

Comment:
Burde vi endre dette til å være vilkårlig antall siffer, i stedet for bare 2, i tilfelle vi legger til noe 128x128-ikoner eller noe?